ludwig/data/preprocessing.py

```diff
@@ -1087,7 +1087,7 @@ def build_metadata(dataset_df, features, global_preprocessing_parameters, backen
             ).get_feature_meta
 
             metadata[feature[NAME]] = get_feature_meta(
-                dataset_df[feature[NAME]].astype(str),
```

Comment:
@w4nderlust do you know which features rely on this conversion for metadata computation?  Can we move it into those features only?

Reply:
Per our conversation: changing so that we only cast to string if the dtype is object.

Reply:
lgtm